### PR TITLE
Bitcoind: ZeroMQ Support + Auto-Update Balance in DesktopApp

### PR DESCRIPTION
**Pull request by alvasw**
Consumers can register listeners when: 
- a new block gets mined
- a UTXO is spent
- a receive address received money

### CODE REVIEW

File: application/src/main/java/bisq/application/DefaultApplicationService.java [hunk 1]
```diff
@@ -137,7 +137,8 @@ public DefaultApplicationService(String[] args) {
         protocolService = new ProtocolService(networkService, identityService, persistenceService, openOfferService);
 
         Optional<WalletConfig> walletConfig = !isRegtestRun() ? Optional.empty() : createRegtestWalletConfig();
-        walletService = new WalletService(walletConfig);
+        walletService = new WalletService(persistenceService, walletConfig);
+        walletService.tryAutoInitialization();
```
Comment (chimp1984) [inline application/src/main/java/bisq/application/DefaultApplicationService.java]: We only create the services in the constructor but any initialisation should happen in the initialize method inside the completable future calls

Comment (chimp1984) [reply application/src/main/java/bisq/application/DefaultApplicationService.java]: Ah just saw you call it already there. So I assume that is a left over and need to be removed.

File: desktop/src/main/java/bisq/desktop/primary/main/content/wallet/receive/WalletReceiveModel.java [hunk 2]
```diff
@@ -22,11 +22,7 @@
...
 
 public class WalletReceiveModel implements Model {
-    private final ObservableList<String> listItems = FXCollections.observableArrayList();
-
-    public void addNewAddress(String address) {
-        listItems.add(address);
-    }
+    final ObservableList<String> listItems = FXCollections.observableArrayList();
 
     public ObservableList<String> getListItems() {
```
Comment (chimp1984) [inline desktop/src/main/java/bisq/desktop/primary/main/content/wallet/receive/WalletReceiveModel.java]: Better use the lombok @Getter for reducing boilerplate

File: wallets/src/main/java/bisq/wallets/bitcoind/zeromq/BitcoindZeroMqTopicProcessors.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+import bisq.wallets.bitcoind.rpc.BitcoindDaemon;
+import bisq.wallets.bitcoind.rpc.responses.BitcoindDecodeRawTransactionResponse;
+import com.google.common.io.BaseEncoding;
+
+import java.util.Set;
+
+public class BitcoindZeroMqTopicProcessors {
+
+    public static final String TOPIC_HASHBLOCK = "hashblock";
```
Comment (chimp1984) [inline wallets/src/main/java/bisq/wallets/bitcoind/zeromq/BitcoindZeroMqTopicProcessors.java]: Wouldn't be a enum more appropriate here?

File: wallets/src/main/java/bisq/wallets/WalletService.java [hunk 4]
```diff
@@ -121,9 +160,21 @@ public boolean isWalletReady() {
...
+
+                listUnspent().thenAccept(utxos -> {
+                    utxoTxIds.clear();
+                    utxos.stream()
+                            .map(Utxo::getTxId)
+                            .forEach(utxoTxIds::add);
+                });
+            }
+
+            return (long) walletBalance;
```
Comment (chimp1984) [inline wallets/src/main/java/bisq/wallets/WalletService.java]: Is walletBalance in satoshis? Weird why they use double then... support for sub-satoshis?

**Review by chimp1984 — CHANGES_REQUESTED** [review]
Looks good overall, just a few small issues and questions.
Have not tested it yet.

File: wallets/src/main/java/bisq/wallets/WalletService.java (hunk 4 above)
Comment (alvasw) [reply wallets/src/main/java/bisq/wallets/WalletService.java]: It's in BTC (see https://developer.bitcoin.org/reference/rpc/getbalance.html).

File: application/src/main/java/bisq/application/DefaultApplicationService.java (hunk 1 above)
Comment (alvasw) [reply application/src/main/java/bisq/application/DefaultApplicationService.java]: Yes, that's a left over. Will fix that.

File: wallets/src/main/java/bisq/wallets/WalletService.java (hunk 4 above)
Comment (chimp1984) [reply wallets/src/main/java/bisq/wallets/WalletService.java]: But then `return (long) walletBalance;` is wrong, no? As well as `observableBalanceAsCoin.get().getValue() != walletBalance`

Comment (alvasw) [reply wallets/src/main/java/bisq/wallets/WalletService.java]: You're correct. I was distracted.

**Review by chimp1984 — APPROVED** [review]
utACK

**Pull request merged**